hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -0,0 +1,188 @@
+  private Properties props;
+
+  public HoodieTableConfig(FileSystem fs, String metaPath) {
+    Properties props = new Properties();
+    Path propertyPath = new Path(metaPath, HOODIE_PROPERTIES_FILE);
+    log.info("Loading dataset properties from " + propertyPath);
+    try {
+      try (FSDataInputStream inputStream = fs.open(propertyPath)) {
+        props.load(inputStream);
+      }
+    } catch (IOException e) {
+      throw new HoodieIOException("Could not load Hoodie properties from " + propertyPath, e);
+    }
+    this.props = props;
+  }
+
+  public HoodieTableConfig(Properties props) {
+    this.props = props;
+  }
+
+  /**
+   * For serailizing and de-serializing
+   *
+   * @deprecated
+   */
+  public HoodieTableConfig() {
+  }
+
+  /**
+   * Initialize the hoodie meta directory and any necessary files inside the meta (including the
+   * hoodie.properties)
+   */
+  public static void createHoodieProperties(FileSystem fs, Path metadataFolder,
+      Properties properties) throws IOException {
+    if (!fs.exists(metadataFolder)) {
+      fs.mkdirs(metadataFolder);
+    }
+    Path propertyPath = new Path(metadataFolder, HOODIE_PROPERTIES_FILE);
+    try (FSDataOutputStream outputStream = fs.create(propertyPath)) {
+      if (!properties.containsKey(HOODIE_TABLE_NAME_PROP_NAME)) {
+        throw new IllegalArgumentException(
+                HOODIE_TABLE_NAME_PROP_NAME + " property needs to be specified");
+      }
+      if (!properties.containsKey(HOODIE_TABLE_TYPE_PROP_NAME)) {
+        properties.setProperty(HOODIE_TABLE_TYPE_PROP_NAME, DEFAULT_TABLE_TYPE.name());
+      }
+      if (properties.getProperty(HOODIE_TABLE_TYPE_PROP_NAME) == HoodieTableType.MERGE_ON_READ
+              .name()
+              && !properties.containsKey(HOODIE_PAYLOAD_CLASS_PROP_NAME)) {
+        properties.setProperty(HOODIE_PAYLOAD_CLASS_PROP_NAME, DEFAULT_PAYLOAD_CLASS);
+      }
+      if (!properties.containsKey(HOODIE_ARCHIVELOG_FOLDER_PROP_NAME)) {
+        properties.setProperty(HOODIE_ARCHIVELOG_FOLDER_PROP_NAME, DEFAULT_ARCHIVELOG_FOLDER);
+      }
+      properties
+              .store(outputStream, "Properties saved on " + new Date(System.currentTimeMillis()));
+    }
+  }
+
+
+  /**
+   * Read the table type from the table properties and if not found, return the default
+   */
+  public HoodieTableType getTableType() {
+    if (props.containsKey(HOODIE_TABLE_TYPE_PROP_NAME)) {
+      return HoodieTableType.valueOf(props.getProperty(HOODIE_TABLE_TYPE_PROP_NAME));
+    }
+    return DEFAULT_TABLE_TYPE;
+  }
+
+  /**
+   * Read the payload class for HoodieRecords from the table properties
+   */
+  public String getPayloadClass() {
+    // There could be datasets written with payload class from com.uber.hoodie. Need to transparently
+    // change to org.apache.hudi
+    return props.getProperty(HOODIE_PAYLOAD_CLASS_PROP_NAME, DEFAULT_PAYLOAD_CLASS)
+        .replace("com.uber.hoodie", "org.apache.hudi");
```

Comment:
@vinothchandar : this is a special change

Reply:
can we create a JIRA and collect all such the special changes, so we can undo in subsequent releases?

Reply:
Done. https://issues.apache.org/jira/browse/HUDI-202